`corepack pnpm@<version>` doesn't work anymore.

### Last pnpm version that worked

v8.15.6

### pnpm version

v9.0.0-beta.2

### Code to reproduce the issue

`corepack pnpm@9.0.0-beta.2 install`

### Expected behavior

It works.

### Actual behavior

`ERR_PNPM_BAD_PM_VERSION  This project is configured to use v8.15.6 of pnpm. Your current pnpm is v9.0.0-beta.2`

### Additional information

Being able to do this is a deliberate feature of Corepack.
> `corepack <binary name>[@<version>] [... args]`
> This meta-command runs the specified package manager in the local folder. You can use it to force an install to run with a given version, which can be useful when looking for regressions.

[Source](https://github.com/nodejs/corepack#utility-commands)

### Node.js version

v18.19.1

### Operating System

Linux

--- a/cli/cli-utils/src/packageIsInstallable.ts
+++ b/cli/cli-utils/src/packageIsInstallable.ts
@@ -23,7 +23,7 @@ export function packageIsInstallable (
   const pnpmVersion = packageManager.name === 'pnpm'
     ? packageManager.version
     : undefined
-  if (pkg.packageManager) {
+  if (pkg.packageManager && !process.env.COREPACK_HOME) {
     const [pmName, pmReference] = pkg.packageManager.split('@')
     if (pmName && pmName !== 'pnpm') {
       const msg = `This project is configured to use ${pmName}`

